user/inc/Arduino.h

```diff
@@ -193,6 +193,9 @@ typedef volatile uint32_t RwReg;
 // C++ only
 #ifdef __cplusplus
 
+#ifndef __clang__
```

Comment:
I would suggest also not using targeting `__clang__` specifically and instead include `<cmath>` if compiling as `__cplusplus` and `math.h` otherwise, which should resolve the problem.

Reply:
Fixed. Note that my fix still includes `math.h` in all cases since I think that's required for backwards compatibility with user code as described in the new comment. My C experience is limited so I'm prepared to be wrong about that!

I also chose to put the conditional include of `<cmath>` adjacent to the include of `math.h` since I think it's clearer even though it requires a separate `#ifndef __cplusplus` guard.